src/HW_models/NRF_RTC.c

```diff
@@ -39,90 +39,142 @@
 #define N_CC 4
 
 #define RTC_COUNTER_MASK 0xFFFFFF /*24 bits*/
+#define RTC_TRIGGER_OVERFLOW_COUNTER_VALUE 0xFFFFF0
+
+#define SUB_US_BITS 9 // Bits representing sub-microsecond units
+
 NRF_RTC_Type NRF_RTC_regs[N_RTC];
 
 static bool RTC_Running[N_RTC] = {false};
 static uint32_t RTC_INTEN[N_RTC] = {0};
 
 bs_time_t Timer_RTC = TIME_NEVER;
 static bs_time_t cc_timers[N_RTC][N_CC] = {{TIME_NEVER}}; //when each CC will match (in microseconds)
+static bs_time_t overflow_timer[N_RTC] = {TIME_NEVER}; //when the timer will overflow (in microseconds)
+
+static bs_time_t overflow_timer_sub_us[N_RTC] = {TIME_NEVER}; //when the timer will overflow (in sub-microsecond units)
+static bs_time_t RTC_counter_startT_sub_us[N_RTC] = {TIME_NEVER}; //Time when the counter was "started" (really the time that would correspond to COUNTER = 0)
+static bs_time_t RTC_counter_startT_negative_sub_us[N_RTC] = {0};
 
-static bs_time_t RTC_counter_startT[N_RTC] = {TIME_NEVER}; //Time when the counter was "started" (really the time that would correspond to COUNTER = 0)
 static uint32_t counter[N_RTC] = {0}; //Internal counter value when the counter was stopped
 
-static bs_time_t first_lf_tick_time = 0;
+static bs_time_t first_lf_tick_time_sub_us = 0;
```

Comment:
I would have used a uint64_t instead of bs_time_t for all the submicroseconds variables (bs_time_t is understood not just as a 64 bit value, but as representing a time in microseconds)

Reply:
Ok, changed everywhere